src/Simplex/Messaging/Agent/Store/SQLite/Schema.hs

```diff
@@ -69,35 +69,128 @@ sndQueues =
+        ON DELETE CASCADE
+        DEFERRABLE INITIALLY DEFERRED
+    ) WITHOUT ROWID;
+  |]
+
+-- table containing metadata of messages agent receives
+-- * external_snd_id
+--   id of the message at sender, i.e. 'external_snd_id' corresponds to 'internal_snd_id' from the sender's side.
+-- * rcv_status
+--   one of [semantically]: "received", "acknowledged to broker", "acknowledged to sender", changed in this order.
+-- * broker_ack_ts
+--   ts of acknowledgement to broker, corresponds to "acknowledged to broker" status, should be null until that;
+--   do not mix up with 'broker_ts' - ts created at broker after broker receives the message from sender.
+-- * sender_ack_ts
+--   ts of acknowledgement to sender, corresponds to "acknowledged to sender" status, should be null until that;
+--   do not mix up with 'external_snd_ts' - ts created at sender before sending (which corresponds to 'internal_ts').
+--
+-- the order of inserting rcv messages - in transaction do:
+-- 1. look up 'last_internal_msg_id' and 'last_internal_rcv_msg_id' from 'connections' table;
+-- 2. increment internal ids and insert into 'messages' table;
+-- 3. insert into 'rcv_messages' table;
+-- 4. update internal ids in 'connections' table - application is responsible for consistency.
+rcvMessages :: Query
+rcvMessages =
+  [sql|
+    CREATE TABLE rcv_messages(
+      conn_alias TEXT NOT NULL,
+      internal_rcv_id INTEGER NOT NULL,
+      internal_id INTEGER NOT NULL,
+      external_snd_id INTEGER NOT NULL,
+      external_snd_ts TEXT NOT NULL,
+      broker_id INTEGER NOT NULL,
+      broker_ts TEXT NOT NULL,
+      rcv_status TEXT NOT NULL,
+      broker_ack_ts TEXT,
+      sender_ack_ts TEXT,
+      PRIMARY KEY (conn_alias, internal_rcv_id),
+      FOREIGN KEY (conn_alias, internal_id)
+        REFERENCES messages (conn_alias, internal_id)
+        ON DELETE CASCADE
+    ) WITHOUT ROWID;
+  |]
+
+-- table containing metadata of messages agent sends
+-- * internal_snd_id
+--   id of the message sent / to be sent, as in its number in order of sending.
+-- * snd_status
+--   one of [semantically]: "created", "sent", "delivered", changed in this order.
+-- * sent_ts
+--   ts of msg received by broker, corresponds to "sent" status, should be null until that.
+-- * delivered_ts
+--   ts of msg received by recipient, corresponds to "delivered" status, should be null until that.
+--
+-- the order of inserting snd messages - in transaction do:
+-- 1. look up 'last_internal_msg_id' and 'last_internal_snd_msg_id' from 'connections' table;
+-- 2. increment internal ids and insert into 'messages' table;
+-- 3. insert into 'snd_messages' table;
+-- 4. update internal ids in 'connections' table - application is responsible for consistency.
+sndMessages :: Query
+sndMessages =
+  [sql|
+    CREATE TABLE snd_messages(
+      conn_alias TEXT NOT NULL,
+      internal_snd_id INTEGER NOT NULL,
+      internal_id INTEGER NOT NULL,
+      snd_status TEXT NOT NULL,
```

Comment:
same question here: do we need this field if we have two ts

Just asking - probably easier to filter with the status field

Reply:
yep, probably we don't need it - but you make a good point it'll be more convenient for filtering

Reply:
I vote let's keep those